lib/lambda_ethereum_consensus/state_transition/misc.ex

```diff
@@ -11,4 +13,57 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
     slots_per_epoch = ChainSpec.get("SLOTS_PER_EPOCH")
     div(slot, slots_per_epoch)
   end
+
+  @doc """
+     Return the shuffled index corresponding to ``seed`` (and ``index_count``).
+  """
+  @spec compute_shuffled_index(SszTypes.uint64(), SszTypes.uint64(), SszTypes.bytes32()) ::
+          SszTypes.uint64()
+  def compute_shuffled_index(index, index_count, seed) do
+    Assertions.assert(index > index_count)
```

Comment:
Here use: 

`unless index > index_count`

Reply:
Based on the implementation, it should throw that's why I used that but I would change it. Thanks

Reply:
In elixir we should never throw except if we have an internal error, something that should never happen. Like in Rust you never want to unwrap except if internal failure.

Hence what you want to do in those case of errors is return something following those lines: `{:error, "[Error message of this specific error]"}` enabling us then to handle the error correctly.